packages/medusa/src/services/payment-collection.ts

```diff
@@ -0,0 +1,161 @@
+  }
+
+  async retrieve(
+    paymentCollectionId: string,
+    config: FindConfig<PaymentCollection> = {}
+  ): Promise<PaymentCollection> {
+    const manager = this.transactionManager_ ?? this.manager_
+    const paymentCollectionRepository = manager.getCustomRepository(
+      this.paymentCollectionRepository_
+    )
+
+    const query = buildQuery({ id: paymentCollectionId }, config)
+    const paymentCollection = await paymentCollectionRepository.findOne(query)
+
+    if (!paymentCollection) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `Payment collection with id ${paymentCollectionId} was not found`
+      )
+    }
+
+    return paymentCollection
+  }
+
+  async create(data: CreatePaymentCollectionInput): Promise<PaymentCollection> {
+    return await this.atomicPhase_(async (transactionManager) => {
+      const paymentCollectionRepository =
+        transactionManager.getCustomRepository(
+          this.paymentCollectionRepository_
+        )
+
+      const paymentCollectionToCreate = paymentCollectionRepository.create({
+        region_id: data.region_id,
+        type: data.type,
+        status: PaymentCollectionStatus.NOT_PAID,
+        currency_code: data.currency_code,
+        amount: data.amount,
+        metadata: data.metadata,
+        created_by: data.created_by,
+        description: data.description,
+      })
+
+      const paymentCollection = await paymentCollectionRepository.save(
+        paymentCollectionToCreate
+      )
+
+      await this.eventBusService_
+        .withTransaction(transactionManager)
+        .emit(PaymentCollectionService.Events.CREATED, paymentCollection)
+
+      return paymentCollection
+    })
+  }
+
+  async update(
+    paymentCollectionId: string,
+    data: DeepPartial<PaymentCollection>
+  ): Promise<PaymentCollection> {
+    return await this.atomicPhase_(async (manager) => {
+      const paymentCollectionRepo = manager.getCustomRepository(
+        this.paymentCollectionRepository_
+      )
+
+      const paymentCollection = await this.retrieve(paymentCollectionId)
+
+      for (const key of Object.keys(data)) {
+        if (key === "metadata" && data.metadata) {
+          paymentCollection[key] = setMetadata(paymentCollection, data.metadata)
+        } else if (isDefined(data[key])) {
+          paymentCollection[key] = data[key]
+        }
+      }
+
+      const result = await paymentCollectionRepo.save(paymentCollection)
+
+      await this.eventBusService_
+        .withTransaction(manager)
+        .emit(PaymentCollectionService.Events.UPDATED, result)
+
+      return result
+    })
+  }
+
+  async delete(
+    paymentCollectionId: string
+  ): Promise<PaymentCollection | undefined> {
+    return await this.atomicPhase_(async (manager) => {
+      const paymentCollectionRepo = manager.getCustomRepository(
+        this.paymentCollectionRepository_
+      )
+
+      const paymentCollection = await this.retrieve(paymentCollectionId)
```

Comment:
todo: My bad, missed this one. `retrieve` will throw in case of a non-existing payment collection, so to be idempotent we should catch and return `void 0`
```suggestion
      const paymentCollection = await this.retrieve(paymentCollectionId).catch(() => void 0)
```

Reply:
nice catch. will update the tests as well.
it's only covering the happy path.